src/foam/core/stdlib.js

```diff
@@ -361,8 +361,8 @@ foam.LIB({
     function clone(o) { return o; },
     function equals(a, b) { return a === b; },
     function compare(a, b) {
-      return ( b === null || b === undefined ) ? 1 :
-          a < b ? -1 : a > b ? 1 : 0;
+      return ( b === null || b === undefined || foam.Object.isInstance(b) ||
+          foam.core.FObject.isInstance(b) ) ? 1 : a < b ? -1 : a > b ? 1 : 0;
```

Comment:
Somewhere in `mlang`, there is a direct call to `foam.Number.compare`, which is why I ended up with all those checks. In the next commit, I have changed the use of `foam.Number.compare` to `foam.util.compare` and replaced it with ` ( ! this.isInstance(b) ) ? 1 : a < b ? -1 : a > b ? 1 : 0;`

The latter part of the ternary is still needed, since we need to return 1 and -1 as required for numbers, not just a boolean value.